### ISSUE TITLE
Question

### ISSUE THREAD

**Issue by DavidKartF1** [leaty/camctl#1]
Hi, I bought a Kraken x62 for my next build and i was wondering if it is possible to control radiator fans with this tool if I connect them to the fan splitter included on the kraken kit or if instead I would have to connect radiator fans to my motherboard. Also I want to thank you for creating this tool as Linux support is very important for me. (Sorry for my bad english).

**Comment by leaty** [comment]
This tool speaks directly with the Kraken hardware via its USB connection - so yes connect them to the fan splitter. You're welcome to notify me if you need any help with it.

**Comment by DavidKartF1** [comment]
Nice, thanks for answering. Now I am wondering if I need to connect the 3 pin connector to a pump header or if it would work just connecting it to a CPU fan header becuase I'm planning to buy a motherboard that doesn't have pump header. I think it will work connecting to CPU fan header because as you said, it is controled via its UBS connectio, but I want to be sure about it.

**Comment by leaty** [comment]
Yes, it makes no difference for camctl. As long as everything is also connected to the kraken inputs, it will work fine. What wouldn't work, is by connecting the fans directly to the motherboard.

By the way, I recommend running `lsusb` and looking for "NZXT". This will show you the vid and pid of your cooler - this is what camctl uses to control it. My Kraken X62 had 1e71:170e. Yours should have the same, if not you'll need to modify those variables in the camctl file and then run install.sh again, or before.

**Comment by DavidKartF1** [comment]
Understood, I haven't installed the Kraken yet, so I will try your tool once I install the Kraken. If there is any issue I will post it here. Thank you.

**Comment by leaty** [comment]
Good good, let me know how it goes.

**Comment by DavidKartF1** [comment]
Today I finally received the kraken. I have tested this tool on Elementary OS but it is not working, my kraken also have 1e71:170e so I do not need to change the original code. When I run sudo camctl --help it returns:
`Traceback (most recent call last):
  File "/usr/bin/camctl", line 3, in <module>
    import usb.core
ImportError: No module named 'usb`

Tomorrow I will test the tool on ArchLinux to check if it works properly.

**Comment by leaty** [comment]
Ah, a simple fix. You're just missing a python3 library. See issue #2 for more information.

TL;DR
`sudo pip3 install pyusb` (unsure if pip3 exists by default on Elementary OS though).

Nice choice of distros btw! 👍 

EDIT: I've added dependencies to the [readme](https://github.com/leaty/camctl/blob/master/README.md) to avoid further confusion. I will however not add the pyusb installation to the camctl installer, this avoids breaking things cross-distro - e.g. people with different python setups may require different commands or perhaps they simply want to make use of virtualenv.

**Comment by DavidKartF1** [comment]
Nice, I was trying to fix it with `sudo pip install pyusb` but it doesn't work. pip3 does not exist by default on elementary OS, after installing pip3 and pyusb the tool works. I have been testing it and in one ocasion it failed when changing pump speed (but then I tried again the same command and worked perfectly), here is the issue:
```
`Detaching kernel driver..
Setting pump speed to 50..
Reattaching kernel driver..
Reattaching kernel driver..
Traceback (most recent call last):
  File "/usr/bin/camctl", line 62, in <module>
    cam.declaim()
  File "/usr/bin/camctl", line 27, in declaim
    self.device.attach_kernel_driver(0)
  File "/usr/local/lib/python3.5/dist-packages/usb/core.py", line 1089, in attach_kernel_driver
    interface)
  File "/usr/local/lib/python3.5/dist-packages/usb/backend/libusb1.py", line 906, in attach_kernel_driver
    _check(self.lib.libusb_attach_kernel_driver(dev_handle.handle, intf))
  File "/usr/local/lib/python3.5/dist-packages/usb/backend/libusb1.py", line 595, in _check
    raise USBError(_strerror(ret), ret, _libusb_errno[ret])
usb.core.USBError: [Errno 2] Entity not found

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/camctl", line 64, in <module>
    cam.declaim()
  File "/usr/bin/camctl", line 27, in declaim
    self.device.attach_kernel_driver(0)
  File "/usr/local/lib/python3.5/dist-packages/usb/core.py", line 1089, in attach_kernel_driver
    interface)
  File "/usr/local/lib/python3.5/dist-packages/usb/backend/libusb1.py", line 906, in attach_kernel_driver
    _check(self.lib.libusb_attach_kernel_driver(dev_handle.handle, intf))
  File "/usr/local/lib/python3.5/dist-packages/usb/backend/libusb1.py", line 595, in _check
    raise USBError(_strerror(ret), ret, _libusb_errno[ret])
usb.core.USBError: [Errno 2] Entity not found`
```

I don't think it is a critical issue so no worries. Thanks for your work, the tool work fine. I guess it will work perfectly on Arch too.

**Comment by leaty** [comment]
I'm glad it worked! 

Note: I think by default `sudo python3 -m pip install pyusb` would work if pip3 doesn't exist.

Thanks for the error log, this is likely a distro-specific issue but no big deal. I'm not exactly sure why this happens for you though, it may just be an older kernel - the error suggests that after detaching the kernel driver and taking control (the Linux kernel by default holds the Kraken "hostage" in my testing) and writing to the usb device (setting fan/pump speeds) it can no longer reattach the driver (give it back to the Linux kernel). I only added the attach feature to reset everything back to normal. Seeing as Linux has no usable drivers for the Kraken however, it doesn't matter if its detached, this also resets on reboot so you may see that error again - however it should still set the speeds regardless so you shouldn't need to run the command twice (good to know if you'll be using it as a startup script).
